zebra/zserv.c

```diff
@@ -267,38 +267,45 @@ static int zserv_write(struct thread *thread)
 		return 0;
 
 	msg = stream_fifo_pop(client->obuf_fifo);
-	stream_set_getp(msg, 0);
-	client->last_write_cmd = stream_getw_from(msg, 6);
-
-	writerv = buffer_write(client->wb, client->sock, STREAM_DATA(msg),
-			       stream_get_endp(msg));
-
-	stream_free(msg);
-
-	switch (writerv) {
-	case BUFFER_ERROR:
-		zlog_warn(
-			"%s: buffer_write failed to zserv client fd %d, closing",
-			__func__, client->sock);
-		/*
-		 * Schedule a delayed close since many of the functions that
-		 * call this one do not check the return code. They do not
-		 * allow for the possibility that an I/O error may have caused
-		 * the client to be deleted.
-		 */
-		client->t_suicide = NULL;
-		thread_add_event(zebrad.master, zserv_delayed_close, client, 0,
-				 &client->t_suicide);
-		return -1;
-	case BUFFER_EMPTY:
-		THREAD_OFF(client->t_write);
-		break;
-	case BUFFER_PENDING:
-		thread_add_write(zebrad.master, zserv_flush_data, client,
-				 client->sock, &client->t_write);
-		break;
+	while (msg) {
```

Comment:
done

Reply:
Actually on second thought, I do not think we should do this.  most of the time we get BUFFER_PENDING, so let's fill up the buffer to write as much as possible instead of stopping at 10 packets.  Since we are not actually writing data this should be ok